service-workers/service-worker/resources/service-worker-recorder.js

```diff
@@ -0,0 +1,204 @@
+'use strict';
+self.ServiceWorkerRecorder = (function() {
```

Comment:
> Maybe some comments explaining what ServiceWorkerRecorder is and what you
> would use it for could be helpful?

Sure thing.

> Also might want to make it explicit that despite the need to pass a "worker"
> to the various client methods, all the workers will record to the same
> database, with no way to distinguish records from different workers (or maybe
> it would make sense to key of "scope" as well to at least get some support
> for multiple recording workers?)

This sounds good to me.

> Also I'm not sure what the benefit is of directing all client requests
> through the service worker rather than just directly accessing IDB? Is that
> just to get a clear linearalized order of operations?

Directing client requests in this way allows for keying on the service worker's
location and (in light of your previous comment) scope.